aptos-move/aptos-vm-types/src/module_and_script_storage/state_view_adapter.rs

```diff
@@ -72,6 +82,62 @@ impl<'s, S: StateView, E: WithRuntimeEnvironment> AptosCodeStorageAdapter<'s, S,
         let storage = adapter.into_unsync_code_storage(runtime_environment);
         Self { storage }
     }
+
+    /// Drains cached verified modules from the code storage, transforming them into format used by
+    /// global caches. Should only be called when the code storage borrows [StateView].
+    pub fn into_verified_module_code_iter(
+        self,
+    ) -> Result<
+        impl Iterator<
+            Item = (
+                ModuleId,
+                Arc<ModuleCode<CompiledModule, Module, AptosModuleExtension>>,
+            ),
+        >,
+        PanicError,
+    > {
+        let state_view = match self.storage.module_storage().byte_storage().state_view {
+            BorrowedOrOwned::Borrowed(state_view) => state_view,
+            BorrowedOrOwned::Owned(_) => {
+                return Err(PanicError::CodeInvariantError(
+                    "Verified modules should only be extracted from borrowed state".to_string(),
+                ))
+            },
+        };
+
+        let mut modules_to_add = vec![];
+        for (key, verified_code) in self
+            .storage
+            .into_module_storage()
+            .into_verified_modules_iter()
```

Comment:
we shouldn't need modules_to_add vec at all, since it's basically mapping one to one-so can create and directly return the desired iterator

Reply:
But then iterator needs to be of items of results? You cannot do `map(|..| do_something(..)?)`?